src/reconcile/rec_visibility.c

```diff
@@ -296,8 +295,8 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
          * check is not required for history store updates as they are implicitly committed. As
          * prepared transaction IDs are globally visible, need to check the update state as well.
          */
-        if (!is_hs_page && (F_ISSET(r, WT_REC_VISIBLE_ALL) ? WT_TXNID_LE(r->last_running, txnid) :
-                                                             !__txn_visible_id(session, txnid))) {
```

Comment:
The other problem is that we need to handle double tombstone in this function.

If we have T@0 -> T@20 -> U@10, the assert `__rec_update_save, 34: onpage_upd != NULL || supd_restore` will fire if T@0 is visible.

Reply:
It doesn't fire before because T@0 is always globally visible but now it is no longer true.